File: Task-3/Ruby/Sub-Task-4.rb
def generator(n)
  a = ""

  (1..n).each do |i|
    a += " " * (n - i) + "*" * (2 * i - 1) + "\n"
  end

  a += "*" * (2 * n - 1) + "\n" if n.odd?

  (n - 1).downto(1) do |i|
    a += " " * (n - i) + "*" * (2 * i - 1) + "\n"
  end

  a
end

input_file = 'input.txt'
output_file = 'output.txt'

n = File.read(input_file).to_i

if n > 0
  result = generator(n)

  File.open(output_file, 'w') do |file|
    file.write(result)
  end
end

File: Task-3/C++/Sub-Task-4.cpp
#include <iostream>
#include <fstream>
#include <string>

int main() {
    std::ifstream inputFile("input.txt");
    std::ofstream outputFile("output.txt");

    int n;
    inputFile >> n;
    if (n % 2 == 0) {
        n++;
    }

    for (int i = 1; i <= n; i += 2) {
        outputFile << std::string((n - i) / 2, ' ') + std::string(i, '*') << std::endl;
    }

    for (int i = n - 2; i >= 1; i -= 2) {
        outputFile << std::string((n - i) / 2, ' ') + std::string(i, '*') << std::endl;
    }

    inputFile.close();
    outputFile.close();

    return 0;
}
